tests/rest/client/test_rooms.py

```diff
@@ -495,8 +495,10 @@ def test_get_state_cancellation(self) -> None:
         )
 
         self.assertEqual(HTTPStatus.OK, channel.code, msg=channel.result["body"])
+        # json_body is defined as JsonDict, but it can be any valid JSON.
```

Comment:
Oops, that's me from #13136. I vaguely remember feeling icky about it at the time---but I think that most use cases expect a JSON object and this was a compromise to avoid a bunch of casts or ignores.

I guess we could have distinct properties like `.json_object` or `.json_array` on the FakeChannel which check the type of the decoded data at runtime? Probably overkill though.

Reply:
Should I add an TODO comment near that or something? Not sure if you want an action out of this!

Reply:
I was just making a passing comment really---I wouldn't object to a TODO though!

Reply:
(Aside: the ability to mark some comments as action items while leaving others unmarked was something I liked about ReviewBoard)

Reply:
> I guess we could have distinct properties like `.json_object` or `.json_array` on the FakeChannel which check the type of the decoded data at runtime? Probably overkill though.

I think a reasonable compromise be to leave `json_body` as a `JsonDict` (to avoid a ton of churn) but add a `json_list`. I've done this in aca8cf6808cb22c7186f6ebc5011790033bcbec3.